modules/runner/src/integrationTest/java/org/apache/ignite/internal/rest/authentication/ItAuthenticationTest.java

```diff
@@ -0,0 +1,310 @@
+
+        // REST is available with valid credentials
+        for (RestNode node : nodes) {
+            assertTrue(isRestAvailable(node.httpAddress(), "admin", "password"));
+        }
+
+        // REST is not available with invalid credentials
+        for (RestNode node : nodes) {
+            assertFalse(isRestAvailable(node.httpAddress(), "admin", "wrong-password"));
+        }
+    }
+
+    private void initCluster(String baseUrl, String initClusterBody) {
+        URI clusterInitUri = URI.create(baseUrl + "/management/v1/cluster/init");
+        HttpRequest initRequest = HttpRequest.newBuilder(clusterInitUri)
+                .header("content-type", "application/json")
+                .POST(BodyPublishers.ofString(initClusterBody))
+                .build();
+
+        HttpResponse<String> response = sendRequest(client, initRequest);
+        assertThat(response.statusCode(), is(200));
+        waitForAllNodesStarted(nodes);
+    }
+
+    private void updateClusterConfiguration(String baseUrl, String login, String password, String configToApply) {
+        URI updateClusterConfigUri = URI.create(baseUrl + "/management/v1/configuration/cluster/");
+        HttpRequest updateClusterConfigRequest = HttpRequest.newBuilder(updateClusterConfigUri)
+                .header("content-type", "text/plain")
+                .header("Authorization", basicAuthenticationHeader(login, password))
+                .method("PATCH", BodyPublishers.ofString(configToApply))
+                .build();
+
+        HttpResponse<String> updateClusterConfigResponse = sendRequest(client, updateClusterConfigRequest);
+        assertThat(updateClusterConfigResponse.statusCode(), is(200));
+    }
+
+    private boolean isRestNotAvailable(String baseUrl, String login, String password) {
+        return !isRestAvailable(baseUrl, login, password);
+    }
+
+    private boolean isRestAvailable(String baseUrl, String login, String password) {
+        URI clusterConfigUri = URI.create(baseUrl + "/management/v1/configuration/cluster/");
+        HttpRequest clusterConfigRequest = HttpRequest.newBuilder(clusterConfigUri)
+                .header("Authorization", basicAuthenticationHeader(login, password))
+                .build();
+        int code = sendRequest(client, clusterConfigRequest).statusCode();
+        if (code == 200) {
+            return true;
+        } else if (code == 401) {
+            return false;
+        } else {
+            throw new IllegalStateException("Unexpected response code: " + code);
+        }
+    }
+
+    private static HttpResponse<String> sendRequest(HttpClient client, HttpRequest request) {
+        try {
+            return client.send(request, BodyHandlers.ofString());
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterEach
+    void tearDown() {
+        nodes.forEach(RestNode::stop);
```

Comment:
I would suggest to use a parallel stream to stop the nodes concurrently

Reply:
done